xc/domain.ml

```diff
@@ -1052,80 +1052,90 @@ let restore (task: Xenops_task.task_handle) ~xc ~xs ~store_domid ~console_domid
 type suspend_flag = Live | Debug
 
 let write_libxc_record (task: Xenops_task.task_handle) ~xc ~xs ~hvm xenguest_path domid uuid fd flags progress_callback qemu_domid do_suspend_callback =
```

Comment:
The diff made it quite hard. Can you refactor this as follows:

- rename the old `write_libxc_record` to `write_libxc_record'`
- define the following:
```ocaml
 let write_libxc_record (task: Xenops_task.task_handle) ~xc ~xs ~hvm xenguest_path domid uuid fd flags progress_callback qemu_domid do_suspend_callback =
finally
		(fun() ->
			if is_upstream_qemu domid
			then qmp_write domid (Qmp.Command(None, Qmp.Xen_set_global_dirty_log true));
			write_libxc_record' task ~xc ~xs ~hvm xenguest_path domid uuid fd flags progress_callback qemu_domid do_suspend_callback
)
		(fun() ->
			if is_upstream_qemu domid
			then qmp_write domid (Qmp.Command(None, Qmp.Xen_set_global_dirty_log false));
		)
```

This will also reduce the changes once you have to revert this PR.